src/lib/transport/ip/tcp_send.c

```diff
@@ -171,6 +171,7 @@ int ci_tcp_sendmsg_fill_pkt(ci_netif* ni, ci_tcp_state* ts,
   /* Initialise and fill a packet buffer from an iovec. */
   int n;
   ci_ip_pkt_fmt* pkt = oo_pkt_filler_next_pkt(ni, &sinf->pf, sinf->stack_locked);
+  int p_netif_locked = ci_netif_is_locked(ni);
```

Comment:
I haven't looked at the rest of this to check the fix as a whole, but this line is definitely wrong. What you are checking here is whether _anyone_ holds the netif lock. What you want to know is whether the _caller_ holds the netif lock. That information is in `sinf->stack_locked`.